tests/test_detectAndMeasure.py

```diff
@@ -93,7 +93,7 @@ def _check_values(self, values, minValue=None, maxValue=None):
             self.assertTrue(np.all(values <= maxValue))
 
     def _setup_detection(self, doApCorr=False, doMerge=False,
-                         doSkySources=False, doForcedMeasurement=False):
+                         doSkySources=False, doForcedMeasurement=False, nSkySources=5, badSourceFlags=[]):
```

Comment:
This is starting to get unwieldy (e.g., you've missed the Parameters on the new ones). I would replace these kwargs with a dictionary that you apply with `config.update(**kwargs)`. That way, you don't have to add new kwargs to this method as you want more config functionality.